packages/pages/src/common/src/function/types.ts

```diff
@@ -0,0 +1,131 @@
+import { GetPath } from "../template/types.js";
+import { PluginEvent } from "../ci/ci.js";
+
+/**
+ * Defines the functions and fields that are available to the serverless function.
+ * @public
+ */
+export interface ServerlessFunctionModule<
+  U extends ServerlessFunctionArgument
+> {
+  /** The exported config function */
+  config?: ServerlessFunctionConfig;
+  /** The exported getPath function */
+  getPath?: GetPath<void>;
+  /** The exported function */
+  default: ExecuteServerlessFunction<U>;
+}
+
+/**
+ * The type definition for serverless function itself.
+ * @public
+ */
+export type ExecuteServerlessFunction<T extends ServerlessFunctionArgument> = (
+  props: T
+) => ServerlessFunctionReturnValue;
+
+/**
+ * The exported `config` function's definition.
+ * @public
+ */
+export interface ServerlessFunctionConfig {
+  /** The name of the serverless function feature. */
+  name?: string;
+  /** The http event. */
+  event?: PluginEvent;
+  /** The function's name */
+  functionName?: string;
+}
+
+/**
+ * The standard return value for a serverless function
+ * @public
+ */
+export interface ServerlessFunctionReturnValue {
+  /** HTTP response body (refer to MDN Web Docs). */
+  body: string;
+
+  /** HTTP response status code (refer to MDN Web Docs). */
+  statusCode: number;
+
+  /** HTTP response headers (refer to MDN Web Docs).  */
+  headers: object;
+}
+
+/**
+ * The argument passed to a serverless function
+ * @public
+ */
+export interface ServerlessFunctionArgument {
+  /** Object containing each query parameter as  */
+  queryParams: { [key: string]: string };
+
+  /** Internal ID of the site branch. */
```

Comment:
Is this comment correct?

Reply:
fixed in the code but needs to be fixed on Hitchhikers too